File: backend/src/main/java/com/skewwhiffy/auraltester/controller/ClefController.java
package com.skewwhiffy.auraltester.controller;

import com.skewwhiffy.auraltester.dto.ClefResponse;
import com.skewwhiffy.auraltester.notation.factory.InternalNotationFactory;
import com.skewwhiffy.auraltester.service.AbcService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@AllArgsConstructor
public class ClefController {
    private final InternalNotationFactory internalNotationFactory;
    private final AbcService abcService;

    @RequestMapping("/api/clef")
    public ClefResponse get(String clef) {
        var clefObject = internalNotationFactory.clef(clef);
        var abc = abcService.getAbc(clefObject).getAbc();
        return new ClefResponse(abc);
    }
}

File: backend/src/main/kotlin/com/skewwhiffy/auraltester/controller/ClefController.kt
package com.skewwhiffy.auraltester.controller

import com.skewwhiffy.auraltester.dto.ClefResponse
import com.skewwhiffy.auraltester.notation.factory.InternalNotationFactory
import com.skewwhiffy.auraltester.service.AbcService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ClefController(private val internalNotationFactory: InternalNotationFactory, private val abcService: AbcService) {
    @GetMapping("/api/clef")
    fun get(clef: String): ClefResponse {
        val clefObject = internalNotationFactory.clef(clef)
        val abc = abcService.getAbc(clefObject).abc
        return ClefResponse(abc)
    }
}
